### PR TITLE
Updates to "Upgrade to Vue3" #926

### PR DESCRIPTION
**Pull request by suxscribe**
Rebase and update of @oplexz Vue3 PR #926 

What have been made:
- Rebased the code
- Split App.vue into multiple components
- Added Pinia store
- Integrated i18n
- Added vue-timeago3
- Updated all layouts and features up to the current state of the master branch

Everything should be functional. Anyway further refactoring and improvements are needed

### CODE REVIEW

**Comment by pheiduck** [comment]
Very nice work @suxscribe!
Sadly Docker build fails.

**Comment by suxscribe** [comment]
👌 on it

**Comment by oplexz** [comment]
Thanks for giving my original PR a second life!

After this PR is (hopefully) merged, I'd suggest to update README.md: the screenshot is slightly outdated, and I feel like there's room for improvement for the text -- maybe spice it up with some emojis here and there.

Keep up the good work!

**Comment by pheiduck** [comment]
Will be part of v14

**Comment by pheiduck** [comment]
Minior optimization would be nice if we could use only one `.gitignore` which include every folder/file to ignore.

File: webui/dist/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <title>WireGuard</title>
+    <link rel="manifest" href="./manifest.json" />
+    <link rel="icon" type="image/png" href="./img/favicon.png" />
+    <link rel="apple-touch-icon" href="./img/apple-touch-icon.png" />
+    <meta
+      name="viewport"
+      content="width=device-width, initial-scale=1, viewport-fit=cover"
+    />
+    <meta name="apple-mobile-web-app-capable" content="yes" />
+    <script type="module" crossorigin src="/assets/index-YnjTemP9.js"></script>
```
Comment (mtricht) [inline webui/dist/index.html:14]: Is it necessary to commit the build files if you also build them in the Dockerfile?

Comment (pheiduck) [reply webui/dist/index.html:14]: No, the folder `/webui/dist` should be added to the **gitignore** as well as other files and folders which generated during build.

**Comment by pheiduck** [comment]
@suxscribe please remove the `dist` folder

**Comment by pheiduck** [comment]
@suxscribe please rebase. I updated the target branch to match with master.

**Comment by pheiduck** [comment]
@suxscribe please rebase. Feature Window is open for v14.

**Comment by suxscribe** [comment]
@pheiduck 👌 will do

**Comment by pheiduck** [comment]
:( something goes wrong during rebase

**Comment by suxscribe** [comment]
Yep, messed up a bit 🤦 Trying to recover

**Comment by suxscribe** [comment]
Rebased. Everything seems to work

**Comment by pheiduck** [comment]
@suxscribe could you rebase?
Or we could start to move to https://nuxt.com/

**Comment by pheiduck** [comment]
And I'm not really a fan of making very big code changes all at once, as it could break existing functionality.

**Comment by suxscribe** [comment]
Rebased 👌

@pheiduck
> And I'm not really a fan of making very big code changes all at once, as it could break existing functionality.

Well, true. This needs to be thoroughly tested at least

Btw, why Nuxt? I haven't used it, but it seems that it's SSR and routing features won't be used

**Comment by kaaax0815** [comment]
This feature is very much needed. maybe creating a big breaking upgrade with this and ipv6 is needed for a "good" upgrade path

**Pull request closed without merging**